google-cloud-tools-plugin/testSrc/com/google/gct/idea/debugger/ui/CloudAttachDialogTest.java

```diff
@@ -80,37 +81,67 @@ public void testErrorWhenNoProjectSelected() {
   }
 
   public void testNoModulesFound() {
+    initDialog();
     mockLoggedInUser();
     selectEmptyProject();
     ValidationInfo error = getValidationError();
 
-    // Errors
     assertNotNull(error);
     assertEquals(SELECT_VALID_PROJECT_WARNING, error.message);
-
-    // Warnings
     assertFalse(warningMessage.isVisible());
     assertFalse(warningHeader.isVisible());
 
-    // Module selector
     assertFalse(moduleSelector.isEnabled());
   }
 
   public void testDebuggableModuleSelected() {
     mockLoggedInUser();
+
+    binding = mock(ProjectDebuggeeBinding.class);
+    when(binding.buildResult(any(Project.class))).thenReturn(new CloudDebugProcessState());
+
+    ProjectRepositoryValidator repositoryValidator = mock(ProjectRepositoryValidator.class);
+    SyncResult syncResult = mockDebuggableSyncResult();
+    when(repositoryValidator.checkSyncStashState()).thenReturn(syncResult);
+
+    initDialog();
+    dialog.setProjectRepositoryValidator(repositoryValidator);
+
     selectProjectWithDebuggableModules();
     ValidationInfo error = getValidationError();
 
-    // Errors
     assertNull(error);
 
-    // TODO: complete these once CloudAttachDialog is further refactored for testability
-    // Warnings
-//    assertFalse(warningMessage.isVisible());
-//    assertFalse(warningHeader.isVisible());
+    assertFalse(warningMessage.isVisible());
+    assertFalse(warningHeader.isVisible());
+
+    assertTrue(moduleSelector.isEnabled());
+  }
+
+  /**
+   * Issue #309 - if there is no module loaded (including no default module)
+   * then this indicates that the async module loading is still in progress. We do not
+   * want to display an error to the user until the thread completes is complete to avoid
+   * the flashing error message
+   */
+  public void testUnknownProjectSelected() {
+    initDialog();
+    mockLoggedInUser();
+    selectInProgressProject();
+
+    ValidationInfo error = getValidationError();
+
+    assertNull(error);
+    assertFalse(moduleSelector.isEnabled());
+    assertNull(moduleSelector.getSelectedItem());
+  }
 
-    // Module selector
-//    assertTrue(moduleSelector.isEnabled());
+  private void initDialog() {
+    dialog = new CloudAttachDialog(this.getProject(), binding);
```

Comment:
I considered that, but given that tearDown() needs to clean up any instantiated dialog I didn't see a clean way to do this without having the dialog be a field. If you see a better pattern for this please let me know.